src/Components/Canvas.jsx

```diff
@@ -296,6 +297,9 @@ export default function Canvas(props) {
     );
   };
 
+  // === Fetch data after login ====
+  useLayoutEffect(handleRestore);
```

Comment:
does a normal `useEffect` work?

Reply:
Generally `useEffect` is better because it won't freeze the app if the function called takes a long time for whatever reason, so unless there's a definite reason to use `useLayoutEffect` I'd rather avoid it

Also, this will run every state update (including *every frame* while you're drawing). This probably should be:
```suggestion
  useLayout(handleRestore, []);
```
The `, []` [tells React to run the function only once, when the component first loads.](https://reactjs.org/docs/hooks-effect.html#tip-optimizing-performance-by-skipping-effects)